v1/plugins/logs/eventBuffer_test.go

```diff
@@ -0,0 +1,348 @@
+		t.Fatal("expected 1 event to be dropped")
+	}
+
+	if int64(len(b.buffer)) != limit {
+		t.Fatalf("buffer size mismatch, expected %d, got %d", limit, len(b.buffer))
+	}
+
+	// drop all events that don't meet the upload size limit anymore
+	droppedCount, errs := b.Reconfigure(limit, rest.Client{}, "", 100)
+	for _, err := range errs {
+		expectedErrorMsg := "upload chunk size (195) exceeds upload_size_limit_bytes (100)"
+		if err == nil {
+			t.Fatal("error expected")
+		} else if err.Error() != expectedErrorMsg {
+			t.Fatalf("expected error %v but got %v", expectedErrorMsg, err.Error())
+		}
+	}
+	if droppedCount != 2 {
+		t.Fatalf("number of dropped event mismatch, expected %d, got %d", 2, droppedCount)
+	}
+	dropped, err = b.Push(newTestEvent(t, "4", false))
+	expectedErrorMsg := "upload chunk size (195) exceeds upload_size_limit_bytes (100)"
+	if err == nil {
+		t.Fatal("error expected")
+	} else if err.Error() != expectedErrorMsg {
+		t.Fatalf("expected error %v but got %v", expectedErrorMsg, err.Error())
+	}
+	if dropped {
+		t.Fatal("expected no events to be dropped")
+	}
+
+	droppedCount, errs = b.Reconfigure(limit, rest.Client{}, "", 196)
+	for _, err := range errs {
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	if droppedCount != 0 {
+		t.Fatalf("number of dropped event mismatch, expected %d, got %d", 0, droppedCount)
+	}
+	dropped, err = b.Push(newTestEvent(t, "5", false))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if dropped {
+		t.Fatal("expected no events to be dropped")
+	}
+	dropped, err = b.Push(newTestEvent(t, "6", true))
+	if !errors.Is(err, droppedNDCache{}) {
+		t.Fatalf("expected error %v but got %v", droppedNDCache{}.Error(), err.Error())
+	}
+	if dropped {
+		t.Fatal("expected no events to be dropped")
+	}
+
+	dropped, err = b.Push(newTestEvent(t, "7", true))
+	if !errors.Is(err, droppedNDCache{}) {
+		t.Fatalf("expected error %v but got %v", droppedNDCache{}.Error(), err.Error())
+	}
+	if !dropped {
+		t.Fatal("expected 1 event to be dropped")
+	}
+
+	if int64(len(b.buffer)) != 2 {
+		t.Fatalf("buffer size mismatch, expected %d, got %d", 2, len(b.buffer))
+	}
+
+	limit = int64(3)
+	droppedCount, errs = b.Reconfigure(limit, rest.Client{}, "", 200)
+	for _, err := range errs {
+		if !errors.Is(err, droppedNDCache{}) && err != nil {
+			t.Fatal(err)
+		}
+	}
+	if droppedCount != 0 {
+		t.Fatalf("number of dropped event mismatch, expected %d, got %d", 0, droppedCount)
+	}
+	dropped, err = b.Push(newTestEvent(t, "8", false))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if dropped {
+		t.Fatal("expected no events to be dropped")
+	}
+
+	// change nothing
+	droppedCount, errs = b.Reconfigure(limit, rest.Client{}, "", 200)
+	for _, err := range errs {
+		if !errors.Is(err, droppedNDCache{}) && err != nil {
+			t.Fatal(err)
+		}
+	}
+	if droppedCount != 0 {
+		t.Fatalf("number of dropped event mismatch, expected %d, got %d", 0, droppedCount)
+	}
+
+	close(b.buffer)
+	events := make([]EventV1, 0, limit)
+	for event := range b.buffer {
```

Comment:
Maybe this should be broken out into a support function, updated to check against a list of expected events, and called after each push?

Reply:
Good idea, added a function to check the buffer state.